storage/remote/queue_manager_test.go

```diff
@@ -617,23 +628,56 @@ func createHistograms(numSamples, numSeries int) ([]record.RefHistogramSample, [
 	return histograms, series
 }
 
+func createFloatHistograms(numSamples, numSeries int) ([]record.RefFloatHistogramSample, []record.RefSeries) {
+	floatHistograms := make([]record.RefFloatHistogramSample, 0, numSamples)
+	series := make([]record.RefSeries, 0, numSeries)
+	for i := 0; i < numSeries; i++ {
+		name := fmt.Sprintf("test_metric_%d", i)
+		for j := 0; j < numSamples; j++ {
+			fh := record.RefFloatHistogramSample{
+				Ref: chunks.HeadSeriesRef(i),
+				T:   int64(j),
+				FH: &histogram.FloatHistogram{
+					Schema:          2,
+					ZeroThreshold:   1e-128,
+					ZeroCount:       0,
+					Count:           2,
+					Sum:             0,
+					PositiveSpans:   []histogram.Span{{Offset: 0, Length: 1}},
+					PositiveBuckets: []float64{float64(i) + 1},
+					NegativeSpans:   []histogram.Span{{Offset: 0, Length: 1}},
+					NegativeBuckets: []float64{float64(-i) - 1},
```

Comment:
Buckets in float histograms are absolute values, so this histogram is invalid. How about we reuse the above `createHistograms` and pass a `floatHistogram` bool, and use ToFloat() where appropriate? The return can have 3 slices, of which 2 are histograms, and depending on which one you ask for, only use that slice.